src/root_mean_square.rs

```diff
@@ -0,0 +1,36 @@
+// http://rosettacode.org/wiki/Averages/Root_mean_square
+
+#![feature(convert)]
+use std::f64;
+
+// The mean is not defined for an empty list, so we must return an Option
+fn rms(list: &[f64]) -> Option<f64> {
+    match list.len() {
+        0 => None,
+        n => {
+            let sum = list.iter().fold(0., |a, &b| a + b * b);
+            Some(f64::sqrt(sum / n as f64))
+        }
+    }
+}
+
+#[cfg(not(test))]
+fn main() {
+    let input:Vec<_> = (1..11).map(|x| x as f64).collect();
+
+    // 6.2048368229954285
+    let rms = rms(input.as_slice()).unwrap();
```

Comment:
I would remove the usage of `as_slice` so this code can run on stable Rust. See below for a possible replacement:

```
let rms = rms(&input).unwrap();
```